core/new-gui/src/app/workspace/component/property-editor/versions-display/versions-display.component.ts

```diff
@@ -38,7 +38,14 @@ export class VersionsListDisplayComponent implements OnInit {
     this.retrieveWorkflowByVersion(<number>this.workflowActionService.getWorkflowMetadata()?.wid, vid)
       .pipe(untilDestroyed(this))
       .subscribe(workflow => {
+        // we need to display the version on the paper but keep the original workflow in the background
+        this.workflowActionService.setTempWorkflow(this.workflowActionService.getWorkflow());
+        // disable persist to DB because it is read only
+        this.workflowActionService.setWorkflowPersistFlag(false);
         this.workflowActionService.reloadWorkflow(workflow);
+        this.workflowVersionService.setDisplayParticularVersion(true);
+        // disable modifications because it is read only
+        this.workflowActionService.disableWorkflowModification();
```

Comment:
ahh, continue to the comment above. now I found the "enter" "particularVersionDisplay" logic here. It seems unnatural to me since the logic of such "particularVersionDisplay" is cut in pieces. Do we have a better way to unify them? If the logic is triggered in between components, some of those logic better to be moved into a service.

Reply:
done